src/NUnitConsole/nunit3-console/ArgumentsFileParser.cs

```diff
@@ -0,0 +1,53 @@
+﻿// ***********************************************************************
+// Copyright (c) 2011 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace NUnit.ConsoleRunner
+{
+    internal class ArgumentsFileParser: IConverter<IEnumerable<string>, IEnumerable<string>>
+    {
+        private static readonly Regex ArgsRegex = new Regex("(\\\"(?<arg>.+?)\\\"|(?<arg>[^\\s\"]+))", RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
@CharliePoole Regarding DI. I try making a code in accordance with the principle SOLID and "test driven" if it possible. I see that the most simplest way to do it is to use DI/IoC/IoC containers. Also it allows to avoid an inheritance and a code duplication, it makes my code simpler and it becomes a source of really unit tests without combinatorial explosion to cover all possible cases. I mean the each "if" statement increases the number of combinations of 2 times, but the each class inheritance makes it many times more. But is is only my view and I do not want to promote it to anyone :)